src/components/manage/Sharing/Sharing.jsx

```diff
@@ -85,183 +80,97 @@ const messages = defineMessages({
+  const history = useHistory();
+  const [search, setSearch] = useState('');
+  const [inherit, setInherit] = useState(props_inherit);
+  const [isClient, setIsClient] = useState(false);
+  const [entries, setEntries] = useState(props_entries);
+  const pathname = props.location.pathname;
 
-  /**
-   * Default properties
-   * @property {Object} defaultProps Default properties.
-   * @static
-   */
-  static defaultProps = {
-    inherit: null,
-    login: '',
-  };
+  useEffect(() => {
+    dispatch(getSharing(getBaseUrl(pathname), search));
+    setIsClient(true);
+  }, [dispatch, pathname, search, isClient]);
 
-  /**
-   * Constructor
-   * @method constructor
-   * @param {Object} props Component properties
-   * @constructs Sharing
-   */
-  constructor(props) {
-    super(props);
-    this.onCancel = this.onCancel.bind(this);
-    this.onChange = this.onChange.bind(this);
-    this.onChangeSearch = this.onChangeSearch.bind(this);
-    this.onSearch = this.onSearch.bind(this);
-    this.onSubmit = this.onSubmit.bind(this);
-    this.onToggleInherit = this.onToggleInherit.bind(this);
-    this.state = {
-      search: '',
-      inherit: props.inherit,
-      entries: props.entries,
-      isClient: false,
-    };
-  }
-
-  /**
-   * Component did mount
-   * @method componentDidMount
-   * @returns {undefined}
-   */
-  componentDidMount() {
-    this.props.getSharing(getBaseUrl(this.props.pathname), this.state.search);
-    this.setState({ isClient: true });
-  }
+  useEffect(() => {
+    dispatch(getSharing(getBaseUrl(pathname), search));
+    toast.success(
+      <Toast
+        success
+        title={intl.formatMessage(messages.permissionsUpdated)}
+        content={intl.formatMessage(messages.permissionsUpdatedSuccessfully)}
+      />,
+    );
+  }, [
+    dispatch,
+    intl,
+    pathname,
+    search,
+    updateRequest.loaded,
+    updateRequest.loading,
+  ]);
 
-  /**
-   * Component will receive props
-   * @method componentWillReceiveProps
-   * @param {Object} nextProps Next properties
-   * @returns {undefined}
-   */
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    if (this.props.updateRequest.loading && nextProps.updateRequest.loaded) {
-      this.props.getSharing(getBaseUrl(this.props.pathname), this.state.search);
-      toast.success(
-        <Toast
-          success
-          title={this.props.intl.formatMessage(messages.permissionsUpdated)}
-          content={this.props.intl.formatMessage(
-            messages.permissionsUpdatedSuccessfully,
-          )}
-        />,
-      );
-    }
-    this.setState({
-      inherit:
-        this.props.inherit === null ? nextProps.inherit : this.state.inherit,
-      entries: map(nextProps.entries, (entry) => {
-        const values = find(this.state.entries, { id: entry.id });
-        return {
-          ...entry,
-          roles: values ? values.roles : entry.roles,
-        };
-      }),
+  useEffect(() => {
+    map(props_entries, (entry) => {
```

Comment:
This should be moved in previous useEffect under the condition. https://github.com/plone/volto/pull/4918/files#r1241902386